internal/motor/create_schema.go

```diff
@@ -0,0 +1,67 @@
+package motor
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/sonr-io/sonr/pkg/client"
+	"github.com/sonr-io/sonr/pkg/tx"
+	st "github.com/sonr-io/sonr/x/schema/types"
+	mt "go.buf.build/grpc/go/sonr-io/motor/api/v1"
+)
+
+func (mtr *MotorNode) CreateSchema(requestBytes []byte) (mt.CreateSchemaResponse, error) {
+	var request mt.CreateSchemaRequest
+	if err := json.Unmarshal(requestBytes, &request); err != nil {
+		return mt.CreateSchemaResponse{}, fmt.Errorf("unmarshal request: %s", err)
+	}
+
+	listFields, err := convertFields(request.Fields)
+	if err != nil {
+		return mt.CreateSchemaResponse{}, fmt.Errorf("process fields: %s", err)
+	}
+	createSchemaMsg := st.NewMsgCreateSchema(&st.SchemaDefinition{
+		Creator: mtr.Address,
+		Label:   request.Label,
+		Fields:  listFields,
+	})
+
+	txRaw, err := tx.SignTxWithWallet(mtr.Wallet, "/sonrio.sonr.schema.MsgCreateSchema", createSchemaMsg)
+	if err != nil {
+		return mt.CreateSchemaResponse{}, fmt.Errorf("sign tx with wallet: %s", err)
+	}
+
+	resp, err := mtr.Cosmos.BroadcastTx(txRaw)
+	if err != nil {
+		return mt.CreateSchemaResponse{}, fmt.Errorf("broadcast tx: %s", err)
+	}
+
+	csresp := &st.MsgCreateSchemaResponse{}
+	if err := client.DecodeTxResponseData(resp.TxResponse.Data, csresp); err != nil {
+		return mt.CreateSchemaResponse{}, fmt.Errorf("decode MsgCreateSchemaResponse: %s", err)
+	}
+
+	return mt.CreateSchemaResponse{
+		Did: csresp.WhatIs.Did,
+	}, nil
+}
+
+func convertFields(fields map[string]mt.CreateSchemaRequest_SchemaKind) ([]*st.SchemaKindDefinition, error) {
+	result := make([]*st.SchemaKindDefinition, len(fields))
+	var i int32
```

Comment:
Do we need this variable?

Reply:
It's being used on line 59 to update the `result` list. I believe populating the array this way is more efficient than using `append` when you know the length of the array ahead of time.